### PR TITLE
Update to Swift 4.2, SwiftPM

### PR DESCRIPTION
**Pull request by ratkins**
I haven't made any substantiative changes to the code here, just updated to Swift 4.2 and SwiftPM (dropping the 3rd party argument parser for the SwiftPM one), and removed the code signing stuff from the .xcodeproj. I want to do this so I can submit a Homebrew formula, which requires building a release version from source without modifying upstream.

### CODE REVIEW

**Comment by jeantil** [comment]
Hi robert, 

first thank you so much for this contribution. The only issue I have with this setup is the absence of code signing. 

Couldn't the formula pull signed binaries from the github releases instead of building on each individual user's machine ?

**Comment by ratkins** [comment]
No problem—thanks for the tool, it's been really handy for us on a couple of occasions now!

According to the [Brew docs](https://docs.brew.sh/Acceptable-Formulae), "Our policy is that formulae in the core tap (homebrew/core) must be open-source and either built from source or produce cross-platform binaries (e.g. Java, Mono). _Binary-only formulae should go to homebrew/cask._"

But, according to the [Rejected Casks FAQ](https://github.com/Homebrew/homebrew-cask/blob/master/doc/faq/rejected_casks.md), the cask may be rejected if "The app is both open-source and CLI-only (i.e. it only uses the binary artifact)".

I *think* that with some extra flair in the Formula, I can convince the Brew Bot to generate a "bottle" from the formula and most users should download that artefact rather than building on their own machine. It won't be signed by Apple ([yet?](https://discourse.brew.sh/t/code-signing-installed-executables/2131)), but the integrity of the binary will be checked by brew install.

File: Formula.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
+class Autokbisw < Formula
+  desc "Automatic keyboard/input source switching for OSX"
+  homepage "https://github.com/ratkins/autokbisw"
```
[the remark this answers is not in the document]
Comment (ratkins) [reply Formula.rb]: Ah yes, of course, that shouldn't have been pushed into this PR—I was just fiddling to get the formula right.

**Comment by ratkins** [comment]
> About commit [21ec907](https://github.com/jeantil/autokbisw/commit/21ec907ae4ea3d816ee070ae1ffc6453b65113ed)
> 
> You have to move the files around to match the swift package expected structure. That is perfectly fine. Unfortunately you also formatted the files using swift formatter in the same commit, this breaks file history which is pretty bad.
> 
> Can I bother you to split the 2 operations ? I really really care about file history especially on projects I don't work on very often, breaking the history makes diagnostic way harder :(

Yup, fair enough. I tried to do this originally but my git-fu is weak, I'll have another crack at this tomorrow.

**Comment by jeantil** [comment]
>Yup, fair enough. I tried to do this originally but my git-fu is weak, I'll have another crack at this tomorrow.

thank you so much

**Comment by jeantil** [comment]
> No problem—thanks for the tool, it's been really handy for us on a couple of occasions now!

makes me unreasonably proud :) 

Regarding the brew stuff : I was unaware of these policies. It still feels weird that you have to drop the crypto signature from the original developper to be included in the main brew repository. Until I can find another way I'll go with your approach. 

thanks again.

**Comment by jeantil** [comment]
just curious: do you know if using the swift package stuff will resolve https://github.com/jeantil/autokbisw/issues/3 ?
(you have probably guessed that I have very limited experience in osx development :) )

**Comment by ratkins** [comment]
Probably? `swift build` writes its artefacts to `.build/release`, and the Homebrew formula installs them from there.

**Pull request closed without merging**